### PR TITLE
store: add tag and id like raft package in log.

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut @BusyJay @disksing @zhangjinpeng1987

### CODE REVIEW

**Comment by ngaut** [comment]
LGTM

File: src/raftstore/store/peer.rs [hunk 1]
```diff
@@ -497,10 +508,9 @@ impl Peer {
     fn transfer_leader(&mut self, peer: &metapb::Peer) {
         metric_incr!("raftstore.transfer_leader");
 
-        info!("transfer leader from {:?} to {:?} at region {}",
-              self.peer,
+        info!("{} {} transfer leader to {:?}",
+            self.tag, self.peer_id(),
```
Comment (BusyJay) [inline src/raftstore/store/peer.rs]: seems not formated properly.

File: src/raftstore/store/peer.rs [hunk 2]
```diff
@@ -203,7 +203,13 @@ impl Peer {
 
         let store_id = store.store_id();
         let sched = store.snap_scheduler();
-        let ps = try!(PeerStorage::new(store.engine(), &region, sched));
+        let tag = format!("[region {}]", region.get_id());
+
+        let mut ps = try!(PeerStorage::new(store.engine(), &region, sched));
+        // TODO: should we pass these in new function?
```
Comment (BusyJay) [inline src/raftstore/store/peer.rs]: +1

File: src/raftstore/store/peer.rs [hunk 3]
```diff
@@ -203,7 +203,13 @@ impl Peer {
 
         let store_id = store.store_id();
         let sched = store.snap_scheduler();
-        let ps = try!(PeerStorage::new(store.engine(), &region, sched));
+        let tag = format!("[region {}]", region.get_id());
```
Comment (BusyJay) [inline src/raftstore/store/peer.rs]: Why not put the peer id into tag too?

File: src/raftstore/store/peer.rs [hunk 4]
```diff
@@ -154,6 +154,8 @@ pub struct Peer {
     // if we remove ourself in ChangePeer remove, we should set this flag, then
     // any following committed logs in same Ready should be applied failed.
     pending_remove: bool,
+
+    pub tag: String,
```
Comment (BusyJay) [inline src/raftstore/store/peer.rs]: Wrap the tag with Arc can reduce memory usage.

File: src/raftstore/store/peer.rs (hunk 3 above)
Comment (siddontang) [inline src/raftstore/store/peer.rs]: keep the same with raft mode.

File: src/raftstore/store/peer.rs (hunk 4 above)
Comment (siddontang) [inline src/raftstore/store/peer.rs]: I don't think this cost too many memory, seem no need.

File: src/raftstore/store/peer.rs (hunk 3 above)
Comment (BusyJay) [inline src/raftstore/store/peer.rs]: I think it's ok for raft to keep the peer id in tag too.

Comment (siddontang) [inline src/raftstore/store/peer.rs]: If ok, we may write log like `info!("{} became leader at term {}", self.tag, self.term);`, and I think it is not clear for user to see this log, who became the leader? the tag? 

But I know that put id in tag can reduce code, and I did it before.
/cc @ngaut

Comment (ngaut) [inline src/raftstore/store/peer.rs]: I prefer to rename it to ctx if we add more fields to tag.

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**